contracts/CDPLeverage.sol

```diff
@@ -0,0 +1,456 @@
+
+                // dust pan condition, where remainingPeth cannot be below 0.005 peth
+                if (sub(remainingPeth,releasedPeth) < 5000000000000000) {
+                    if (remainingDebt == 0) {
+                        releasedPeth = remainingPeth;
+                    } else {
+                        releasedPeth = sub(remainingPeth,5100000000000000);
+                        releaseWeth(sender.cdpID, releasedPeth);
+                        remainingPeth = 5100000000000000;
+                        break;
+                    }
+                }
+
+               releaseWeth(sender.cdpID, releasedPeth);
+               remainingPeth -= releasedPeth;
+            }
+
+            // release any final peth
+            uint256 finalPeth;
+            if (remainingDebt == 0 && remainingPeth != 0) {
+                finalPeth = tub.ink(sender.cdpID);
+                releaseWeth(sender.cdpID, finalPeth);
+            } else {
+                finalPeth = 0;
+            }
+
+            // convert left over dai to weth
+            wethAmount = marketBuy(weth, dai, remainingDai);
+
+            payout = add(add(add(releasedPeth,wethAmount),finalPeth),excessWeth);
+        }
+
+        // convert WETH to ETH
+        IWETH(weth).withdraw(payout);
+
+        // give cdp to owner
+        // if dustpan condition hit, cdp will have 0.0051 peth left
+        tub.give(sender.cdpID, msg.sender);
+
+        msg.sender.transfer(payout);
+
+       // delete the sender information from investorAddresses array
+        deleteEntity(msg.sender);
+
+        emit ClosePosition(msg.sender, currPriceEth);
+
+    }
+
+
+    /**
+    * @dev Returns the variables contained in the Investor struct for a given address
+    * @param _addr - investor address
+    */
+    function getInvestor(address _addr)
+        public
+        constant
+        returns (
+          uint _layers,
+          uint principal,
+          uint collatRatio,
+          bytes32 cdpID,
+          uint256 purchPrice,
+          uint daiAmountFinal,
+          uint256 totalDebt,
+          uint256 priceFloor,
+          uint256 index)
+    {
+       Investor memory investor = investors[_addr];
+       return (
+         investor.layers,
+         investor.principal,
+         investor.collatRatio,
+         investor.cdpID,
+         investor.purchPrice,
+         investor.daiAmountFinal,
+         investor.totalDebt,
+         investor.priceFloor,
+         investor.index);
+    }
+
+
+    /**
+    * @dev Returns amount of PETH for a given amount of WETH at exchange rate
+    * @param _wethAmount - literal
+    */
+    function weth2peth(uint256 _wethAmount)
+        view
+        public
+        returns (uint pethAmount) {
+        pethAmount = rdiv(_wethAmount, wmul(tub.gap(), tub.per())) - 1;      // WAD
+    }
+
+
+    /**
+    * @dev Executes a market buy order and returns the amount of asset recieved
+    * @param _to - token getting
```

Comment:
Can we change _to and _from to something more descriptive like buying/selling, I'm finding them kinda confusing in this context